### DISCUSSION TITLE
Python return values on error

### DISCUSSION THREAD

**Discussion by sykhro** [biojppm/rapidyaml#262] [Q&A]
I'm using ryml on Python.
On failure (non-existant node) the API returns -1. But the values returned by the index API are unsigned, so I actually get `18446744073709551615`.
Am I missing something?

**Comment by biojppm** [comment]
Can you give a concrete example?

**Reply by sykhro** [reply]
I'm parsing yml from test262.
```
---
es5id: 10.6-10-c-ii-1-s
description: >
    arguments[i] remains same after changing actual parameters in
    strict mode
flags: [onlyStrict]
---
```
Sometimes, this yaml fragment includes a key called `negative` which indicates something that I must handle in my program.
So what I do is
```
tree = ryml.parse_in_arena(THAT_YAML_STRING)
is_negative_key_present = tree.find_child(0, b"negative") != ctypes.c_uint64(-1).value
```
I must compare the result with `ctypes.c_uint64(-1).value` because `find_child()` is going to return `18446744073709551615`

**Reply by biojppm** [reply]
Apologies for the late reply, github's notifications are broken for me.

Indeed on the C++ side, ryml uses `size_t(-1)` to signify a null index. This is unsigned, and its value will be different in 32bit and 64bit, and it will be the maximum value of `size_t`. This symbol is named `ryml::NONE`, and the value you see is on a 64bit environment. So note that in python you should compare against `ryml.NONE` instead of the actual value, because that value may differ depending on the running environment.
